File: jan-09-25/sub.cpp
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    int prefixCount(vector<string>& words, string pref) {
        int n = pref.size(), wl, isans;
        int ans = 0;
        for (const string& word: words) {
            wl = word.size();
            if (wl >= n) {
                isans = 1;
                for (int j = 0; j < n; j++) {
                    if (word[j] != pref[j]) {
                        isans = 0;
                        break;
                    }
                }
                ans += isans;
            }
        }

        return ans;
    }
};

int main(){

}

File: jan-10-25/sub.py
from typing import List
from collections import defaultdict,Counter

class Solution:
    def wordSubsets(self, words1: List[str], words2: List[str]) -> List[str]:
        
        w2_count = defaultdict(int)
        a = ord('a')

        for word in words2:
            count_mp = Counter(word)
            
            for l,cnt in count_mp.items():
                w2_count[l] = max(w2_count[l],cnt)

        ans = []
        w2_count = w2_count.items()

        for word in words1:
            isValid = True
            count_mp = Counter(word)
            
            for l,cnt in w2_count:
                if count_mp[l] < cnt:
                    isValid = False
                    break
            
            if isValid:
                ans.append(word)

        return ans